File: src/main/java/de/mikeyllp/miniGamesV4/storage/ToggleInvitesStorage.java
package de.mikeyllp.miniGamesV4.storage;

import de.mikeyllp.miniGamesV4.MiniGamesV4;
import de.mikeyllp.miniGamesV4.database.Database;
import org.bukkit.Bukkit;
import org.bukkit.entity.Player;

import java.util.UUID;
import java.util.function.Consumer;

import static de.mikeyllp.miniGamesV4.utils.MessageUtils.sendCustomMessage;

public class ToggleInvitesStorage {

    private final Database db;

    public ToggleInvitesStorage(Database db) {
        this.db = db;
    }

    public void checkToggle(Player player, Consumer<Boolean> callback) {
        UUID playerUuid = player.getUniqueId();
        db.getToggleAsync(playerUuid).thenAccept(toggle -> {
            // This is a short way to check if its 0 or 1
            boolean result = toggle == 0;
            Bukkit.getScheduler().runTask(MiniGamesV4.getInstance(), () -> {
                callback.accept(result);
            });
        });
    }


    //Adds or removes the player from the toggle list
    public void addToggle(Player player) {

        UUID playerUuid = player.getUniqueId();
        db.getToggleAsync(playerUuid).thenAccept(toggle -> {
            // This is if or else in a single line
            int newToggle = (toggle == 1) ? 0 : 1;

            db.setToggleAsync(playerUuid, newToggle).thenRun(() -> {
                Bukkit.getScheduler().runTask(
                        MiniGamesV4.getInstance(),
                        () -> {
                            if (newToggle == 1) {
                                sendCustomMessage(player, "<color:#00E5E5>Du kannst jetzt nicht mehr eingeladen werden!</color:#00E5E5>");
                            } else {
                                sendCustomMessage(player, "<color:#00E5E5>Du kannst jetzt wieder eingeladen werden!</color:#00E5E5>");
                            }
                        });
            });
        });
    }
}

File: src/main/java/de/mikeyllp/miniGamesV4/storage/ToggleInvitesStorage.kt
package de.mikeyllp.miniGamesV4.storage

import de.mikeyllp.miniGamesV4.MiniGamesV4
import de.mikeyllp.miniGamesV4.database.Database
import de.mikeyllp.miniGamesV4.utils.MessageUtils
import org.bukkit.Bukkit
import org.bukkit.entity.Player
import java.util.function.Consumer

class ToggleInvitesStorage(private val db: Database) {
    fun checkToggle(player: Player, callback: Consumer<Boolean>) {
        val playerUuid = player.uniqueId
        db.getToggleAsync(playerUuid).thenAccept(Consumer { toggle: Int ->
            // This is a short way to check if its 0 or 1
            val result = toggle == 0
            Bukkit.getScheduler().runTask(MiniGamesV4.Companion.instance!!, Runnable {
                callback.accept(result)
            })
        })
    }


    //Adds or removes the player from the toggle list
    fun addToggle(player: Player) {
        val playerUuid = player.uniqueId
        db.getToggleAsync(playerUuid).thenAccept(Consumer { toggle: Int? ->
            // This is if or else in a single line
            val newToggle = if (toggle == 1) 0 else 1
            db.setToggleAsync(playerUuid, newToggle).thenRun(Runnable {
                Bukkit.getScheduler().runTask(
                    MiniGamesV4.Companion.instance!!,
                    Runnable {
                        if (newToggle == 1) {
                            MessageUtils.sendMessage(
                                player,
                                "<color:#00E5E5>Du kannst jetzt nicht mehr eingeladen werden!</color:#00E5E5>"
                            )
                        } else {
                            MessageUtils.sendMessage(
                                player,
                                "<color:#00E5E5>Du kannst jetzt wieder eingeladen werden!</color:#00E5E5>"
                            )
                        }
                    })
            })
        })
    }
}
